packages/gotrue/lib/src/types/mfa.dart

```diff
@@ -4,23 +4,33 @@ class AuthMFAEnrollResponse {
   /// ID of the factor that was just enrolled (in an unverified state).
   final String id;
 
-  /// Type of MFA factor. Only `[FactorType.totp] supported for now.
+  /// Type of MFA factor. Supports both `[FactorType.totp]` and `[FactorType.phone]`.
   final FactorType type;
 
-  /// TOTP enrollment information.
-  final TOTPEnrollment totp;
+  /// TOTP enrollment information (only present when type is totp).
+  final TOTPEnrollment? totp;
```

Comment:
I know this isn't great, but I think we can call it that it's a fix. Open to suggestions to avoid this though.

Reply:
See my comment in https://github.com/supabase/supabase-flutter/pull/1188/files#r2131807418